apisix/plugins/datadog.lua

```diff
@@ -80,25 +107,32 @@ local function generate_tag(entry, const_tags)
         tags = {}
     end
 
-    if entry.route_id and entry.route_id ~= "" then
-        core.table.insert(tags, "route_name:" .. entry.route_id)
+    if entry.constant_tags and #entry.constant_tags > 0 then
+        for _, tag in pairs(entry.constant_tags) do
```

Comment:
should use ipairs to iterate over the array.

Reply:
Right, not sure why I used `pairs` here. Fixed: https://github.com/apache/apisix/compare/004b574382cc8dd8f75b4e1693face763da8f730..3d61e5984e58e121478bfcad35c033cdd90f2c12